File: Sunrin-ICPC-2022/Pre-B/PB.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

function isLeaf(y){
    return y % 4 == 0 && y % 100 != 0 || y % 400 == 0;
}

function getMaxDate(y, m){
    if(m == 2) return 28 + (isLeaf(y) ? 1 : 0);
    else if(m <= 7 && m % 2 == 1 || m >= 8 && m % 2 == 0) return 31;
    else return 30;
}

const T = parseInt(input[0]);
for(let i=1; i<=T; i++){
    const line = input[i].split(' ');
    const y = parseInt(line[0]);
    const m = parseInt(line[1]);
    if(m > 1) console.log(y + ' ' + (m-1) + ' ' + getMaxDate(y, m-1));
    else console.log(y-1 + ' 12 31');
}

File: Sunrin-ICPC-2022/Pre-B/PB.py
def is_leaf(y):
    return (y % 4 == 0 and y % 100 != 0) or y % 400 == 0

def get_max_date(y, m):
    if m == 2: return 28 + (1 if is_leaf(y) else 0)
    elif (m <= 7 and m % 2 == 1) or (m >= 8 and m % 2 == 0): return 31
    else: return 30

T = int(input())
for tc in range(T):
    y, m = map(int, input().split())
    if m > 1: print(y, m-1, get_max_date(y, m-1))
    else: print(y-1, 12, 31)